### PR TITLE
PP-8277 add state for switching psp

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- add a new state for switching payment provider

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/pact/ContractTest.java [hunk 1]
```diff
@@ -522,4 +525,39 @@ public void aWorldpayGatewayAccountWithFilledCredentialsWithIdExists() {
                 .withPaymentGateway(WORLDPAY.getName())
                 .build());
     }
+
+    @State("a Worldpay gateway account with id 444 with two credentials ready to be switched")
+    public void a() {
```
Comment (nimalank7) [inline src/test/java/uk/gov/pay/connector/pact/ContractTest.java]: What's the full method name?

**Review by nimalank7 — CHANGES_REQUESTED** [review]
What's the full method name?

File: src/test/java/uk/gov/pay/connector/pact/ContractTest.java [hunk 2]
```diff
@@ -522,4 +525,39 @@ public void aWorldpayGatewayAccountWithFilledCredentialsWithIdExists() {
...
+                        .withProviderSwitchEnabled(true)
+                        .build());
+
+        AddGatewayAccountCredentialsParams activeParams =
+                AddGatewayAccountCredentialsParams.AddGatewayAccountCredentialsParamsBuilder
+                        .anAddGatewayAccountCredentialsParams()
+                        .withGatewayAccountId(Long.valueOf(gatewayAccountId))
+                        .withCredentials(Map.of())
+                        .withExternalId(activeExtId)
+                        .withState(ACTIVE)
```
Comment (nimalank7) [inline src/test/java/uk/gov/pay/connector/pact/ContractTest.java:548]: Does this need `.withPaymentProvider("worldpay")`?

**Review by nimalank7 — CHANGES_REQUESTED** [review]
One minor comment

File: src/test/java/uk/gov/pay/connector/pact/ContractTest.java (hunk 2 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/pact/ContractTest.java:548]: Not really, we don't check what the current provider is. But I'll add something

Comment (nimalank7) [reply src/test/java/uk/gov/pay/connector/pact/ContractTest.java:548]: Cool!

**Review by nimalank7 — DISMISSED** [review]
LGTM!

**Review by nimalank7 — APPROVED** [review]
LGTM!

**Pull request merged**